docs/azureai/ai-integrations-compatibility-matrix.md

```diff
@@ -0,0 +1,196 @@
+
+| **Hosting Integration**                    | **Aspire.OpenAI**            | **Aspire.Azure.AI.OpenAI**                    | **Aspire.Azure.AI.Inference** |
+|--------------------------------------------|------------------------------|-----------------------------------------------|-------------------------------|
+| `Aspire.Hosting.Azure.AI.Foundry`          | ❌ No                       | ⚠️ Partial                                    | ✅ Yes (preferred)           |
+| `Aspire.Hosting.Azure.CognitiveServices`   | ❌ No                       | ✅ Yes (preferred)                            | ❌ No                        |
+| `Aspire.Hosting.OpenAI`                    | ✅ Yes (preferred)          | ✅ Yes                                        | ❌ No                        |
+| `Aspire.Hosting.GitHub.Models`              | ⚠️ Partial                  | ❌ No                                         | ✅ Yes (preferred)           |
+
+### Legend
+
+- ✅ **Yes**: The client integration is compatible with the hosting integration
+- ⚠️ **Partial**: The client integration might not support all models or APIs
+- ❌ **No**: The client integration does not support the hosting environment
+- **(preferred)**: The recommended client integration for this hosting integration
+
+## Recommended pairings
+
+In general, use **Aspire.Azure.AI.Inference** to connect to Azure hosted models, or **Aspire.OpenAI** to connect directly to OpenAI.
+
+### Azure AI Foundry
+
+For Azure AI Foundry resources, use the **Aspire.Azure.AI.Inference** client integration. This provides the best compatibility with the diverse range of models available through Azure AI Foundry.
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var foundry = builder.AddAzureAIFoundry("foundry");
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(foundry);
+```
+
+In your service, use the Azure AI Inference client:
+
+```csharp
+builder.AddAzureChatCompletionsClient("foundry");
+```
+
+### Azure Cognitive Services (Azure OpenAI)
+
+For Azure OpenAI resources, use the **Aspire.Azure.AI.OpenAI** client integration for full Azure-specific features and authentication support.
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var openai = builder.AddAzureOpenAI("openai");
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(openai);
+```
+
+In your service, use the Azure OpenAI client:
+
+```csharp
+builder.AddAzureOpenAIClient("openai");
+```
+
+### Direct OpenAI
+
+For direct OpenAI API access, use the **Aspire.OpenAI** client integration.
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var openai = builder.AddOpenAI("openai");
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(openai);
+```
+
+In your service, use the OpenAI client:
+
+```csharp
+builder.AddOpenAIClient("openai");
+```
```

Comment:
Same problem as above, we need to differentiate between the hosting and client integration. Whenever we show `DistributedApplication` we're talking about the hosting integration, and we should include the NuGet package that's related to the code shown. Then we should say something about clients that consume said resources.